### ISSUE TITLE
Respect JVM platform.config settings 

### ISSUE THREAD

**Issue by pkolaczk** [scalacenter/bloop#968]
It looks like JDK home and other JVM settings are currently ignored.
This breaks multi-JDK projects.

I have this in my json config:
```        
        "platform" : {
            "name" : "jvm",
            "config" : {
                "home" : "/opt/jdk11",
                "options" : [
                    "-Xms2g",
                    "-Xmx2g"
                ]
            },
            "mainClass" : [
            ]
        }
```

but bloop still compiles this project with JDK8.

**Comment by pkolaczk** [comment]
@jvican Any hints how to fix it?

**Comment by pkolaczk** [comment]
Is it this one that creates the java compiler to compile the project?
https://github.com/scalacenter/bloop/blob/1fcd2e8359e9b54f896d919f6b19f04b427aff64/backend/src/main/scala/bloop/CompilerCache.scala#L94

**Comment by jvican** [comment]
@pkolaczk I'll have a look at this myself, we need some changes in our CI to test this properly and the fix is required in several other places. I've heard we're not using java home for running/testing an application, but we're indeed using it for compilation. I'll make sure that's the case in my PR.

**Comment by pkolaczk** [comment]
What is the timeline for the full fix for that?
Maybe can you only test my fix just for regressions and merge it so people don't have to patch manually?(which is quite cumbersome process - need to checkout master from my repo, rebuild and reinstall - the most complex step of the installation). 

I fixed the performance regression problem and now the system compiler is used when you use the same java version as the one running Bloop.

**Comment by jvican** [comment]
This issue will be fixed before the 1.3.3 release which I want to release next Thursday.

**Comment by pkolaczk** [comment]
Thank you. Let me know when you merge it. I'll test! :)
